lib/jbuilder/jbuilder_template.rb

```diff
@@ -49,11 +49,12 @@ def initialize(context, options = nil)
   #
   #   json.comments @post.comments, partial: "comments/comment", as: :comment, cached: true
   #
-  def partial!(*args)
-    if args.one? && _is_active_model?(args.first)
-      _render_active_model_partial args.first
+  def partial!(partial_or_model = nil, **options)
+    if partial_or_model.class.respond_to?(:model_name) && partial_or_model.respond_to?(:to_partial_path)
+      _render_active_model_partial partial_or_model
     else
-      _render_explicit_partial(*args)
+      options[:partial] = partial_or_model if partial_or_model
+      _render_partial_with_options options
```

Comment:
`_render_explicit_partial` does a bunch of work to setup the options, but ultimately all it does is add this `partial` key. `_render_partial_with_options` already does a bunch of work to manage the options, so the original implementation was more or less doing the same work twice. See below.

Going back through the history, `_render_explicit_partial` existed before the custom `Jbuilder::CollectionRenderer` was implemented. At that point in time it may have made sense to do this because this would have had to have handled collections, but `_render_partial_with_options` now does all of that busy work.

Reply:
This also saves on an array allocation from the `*args` splat.